combat-trainer.lic

```diff
@@ -259,6 +284,22 @@ class SetupProcess
     game_state.update_whirlwind_weapon_info(offhand_skill)
     game_state.wield_whirlwind_offhand
   end
+  
+  def regalia_shutdown(game_state)
+    bput('retreat', 'You are already as far away as you can get', 'You retreat from combat', 'You sneak back out of combat', 'retreat', 'You try to back ', 'sneak', 'grip remains solid', 'You must stand first', 'grip on you', 'You stop advancing') if Flags['ct-engaged']
```

Comment:
❌ Please use `DRCT.retreat` in place of this `bput` (and the others in this file)

Reply:
What you're seeing there is actually the bput match from DRCT.retreat.

As far as I can see, DR common-travel's retreat retreats until it fully leaves combat.  Problem is,  trying this with plate sticks me in a retreat loop that lasts for 10-15 minutes until regalia expires naturally, even back-training against mobs half my rank.  Brigandine has slightly less severe problems, but can still lose a couple minutes against mobs at my level. I was unable to find a common function that was more forgiving.

I'm on board with adding a gamestate helper function for the switchoff to reduce the number of times this is repeated, but if it's considered bad form to do a retreat that accepts failure then I may either remove retreating entirely or add a new setting for retreating before swapping.  I'll think about it a bit first unless someone knows of an existing common function.  

Maybe have that function retreat twice for brig and plate but use DRCT.retreat for light and chain?  Is that too messy?

Reply:
As far as I'm aware, if you want to retreat, you generally want to fully retreat.

I'm curious about this plate/brig situation. Are you saying the first retreat is easier than the second in plate?

Reply:
So my exact problem is this:  Picture me with ~120 ranks in just about anything defensive/offensive/armors.  I have light encumbrance from the plate, but a lot of hindrance.  I'm testing this in, say endrus_serpents, so I'm very overpowered for the scenario.  I'm wearing full plate and it's time for CT to swap to, say, training Light armor.  The plate has 8 minutes left on it, so I need to retreat and remove the existing Regalia, before I can cast the new Regalia in its place.

At that moment I start retreating and I get stuck in retreat -> 'You try to retreat but are unable to get away.'  I never do anything else except spam retreat every 3 seconds, and I don't make it out past pole range for the next 8 minutes until the plate expires and disappears, at which point I dodge, buck naked, right out of combat and switch to the next regalia.  It's a poor and inefficient way to spend a hunt.  To be fair, I think if Traders were meant to wear full-plate they'd have made us Paladins, but the allure of TDPs is too darn high.

With the 'just try twice' method, I get out of combat wearing chain or light armor, but with brigandine I may only get out to pole range.  It's fine though, there's no RT for removing Regalia pieces, so it just spams through them, and the cast time for Regalia is only like 6 seconds or something.

Keep in mind, preexisting armor-swapping in combat-trainer doesn't retreat at all.  Even if you're swapping out everything on your body it just sits in combat while it gets naked.

I can solve this in two obvious ways.  One is to destroy the armor before retreating (this has no RT, so can spammed without any delay and immediately followed by retreat), then zip out of combat naked with no hindrance, and prepare+cast the regalia spell (~6 seconds?) before resuming the fight. The other way is to handle retreating like I'm doing now, (try it a couple times then give up) or by making a few adjustments to handle plate specifically.  The first option is neater so I'm leaning towards that.  I will poll some other traders to see what they want -- I just don't want to do something stupid in the code.

Reply:
I can see you've done your due diligence and I trust you. Talk to the other traders and let me know what you feel is best.

Reply:
Thanks Robert, and this has been very helpful.

After looking closer I was able to determine I was wrong about the order of events, and swapping cast orders was one of the commits I made mid-update.  None of the regalia swaps/refreshes cause any RT, so there's only one case where there's any RT between swapping sets, at the very start when it puts on your regalia gearset.  I've adjusted that step to use DRCT retreat and the others have had the retreats removed and all call a common function to break their regalia.  I think that's the best of both worlds.

I want to run it through a couple hunting cycles as per usual, but after that I'll submit the changes again.  Thanks again for the advice.